src/functions/postTestResults.ts

```diff
@@ -2,51 +2,30 @@ import {TestResultsDAO} from "../models/TestResultsDAO";
 import {TestResultsService} from "../services/TestResultsService";
 import {HTTPResponse} from "../models/HTTPResponse";
 import { MESSAGES } from "../assets/Enums";
-import { ISubSeg } from "../models/ISubSeg";
-/* workaround AWSXRay.captureAWS(...) call obscures types provided by the AWS sdk.
-https://github.com/aws/aws-xray-sdk-node/issues/14
-*/
-/* tslint:disable */
-let AWS:any;
-if (process.env._X_AMZN_TRACE_ID) {
-  /* tslint:disable */
-   AWS = require("aws-xray-sdk");
-} else {
-  console.log("Serverless Offline detected; skipping AWS X-Ray setup");
-}
-/* tslint:enable */
+import { MappingUtil } from "../utils/mappingUtil";
 
-export const postTestResults = async (event: { body: any; }) => {
-  let subseg: ISubSeg | null = null;
-  if (process.env._X_AMZN_TRACE_ID) {
-  const segment = AWS.getSegment();
-  AWS.capturePromise();
-  if (segment) {
-    subseg = segment.addNewSubsegment("postTestResults");
-  }}
+export async function postTestResults(event: { body: any; }) {
+  const subseg = MappingUtil.getSubSegment("postTestResults");
   const testResultsDAO = new TestResultsDAO();
   const testResultsService = new TestResultsService(testResultsDAO);
 
   const payload = event.body;
 
-  try {
-    if (!payload) {
+  if (!payload) {
       if (subseg) { subseg.addError(MESSAGES.INVALID_JSON); }
```

Comment:
If you extract this portion from the try/catch, the subsegment will not be closed

Reply:
There is a finally block which will always execute so we don't need another try.

Reply:
I saw how the finally block works for the other functions but in this case it will not reach the finally block if the payload is empty. Or am I wrong?

Reply:
Finally block are always executed regardless of a try catch.

Reply:
Unless you return before entering in the try/catch block. In this case:
`return Promise.resolve(new HTTPResponse(400, MESSAGES.INVALID_JSON));`

Reply:
Good spot. Why not add the above if condition in this try block. Doesn't make sense to add another try block on top.

Reply:
Agree, that should be the best solution